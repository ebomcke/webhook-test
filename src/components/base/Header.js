import React from 'react';
import { Menu } from 'semantic-ui-react';
import { compose } from 'recompose';
import { withFirebase } from '../../contexts/Firebase';
import { withNavigation } from '../../contexts/Navigation';

const Header = ({ navigate, firebase, responseName }) => (
  <Menu stackable secondary size="huge">
    <Menu.Menu position="right">
      <Menu.Item
        as="a"
        name="Home"
        active={responseName === 'Home'}
        onClick={() =>
          navigate({
            name: 'Home',
          })
        }
      />
      <Menu.Item
        as="a"
        name="Endpoints"
        active={responseName === 'Endpoints'}
        onClick={() =>
          navigate({
            name: 'Endpoints',
          })
        }
      />
      <Menu.Item as="a" name="Logout" onClick={() => firebase.auth().signOut()}>
        Logout
      </Menu.Item>
    </Menu.Menu>
  </Menu>
);

export default compose(
  withNavigation,
  withFirebase,
)(Header);
