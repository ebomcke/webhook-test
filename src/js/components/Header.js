import React, { Component } from 'react'
import './Header.css'

class Header extends Component {
    render() {
        return <div className="header">This is the header</div>
    }
}

export default Header