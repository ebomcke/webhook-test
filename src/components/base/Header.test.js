import React from 'react';
import Header from './Header';
import renderer from 'react-test-renderer';

describe('Rendering tests', () => {
  test('No active link', () => {
    const header = renderer.create(<Header />);
    let tree = header.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Home link active', () => {
    const header = renderer.create(<Header responseName={'Home'} />);
    let tree = header.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Endpoints link active', () => {
    const header = renderer.create(<Header responseName={'Endpoints'} />);
    let tree = header.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Interactions tests', () => {
  const navigate = jest.fn();
  const signOut = jest.fn();
  const firebase = {
      auth: () => ({
          signOut
      })
  };
  const header = renderer.create(
    <Header responseName={'Home'} navigate={navigate} firebase={firebase} />,
  );

  afterEach(() => {
    navigate.mockClear();
  });

  test('Click on Home link navigates to Home view', () => {
    const homeLink = header.root.findByProps({ name: 'Home' });
    homeLink.props.onClick();

    expect(navigate.mock.calls.length).toBe(1);
    expect(navigate.mock.calls[0][0]).toEqual({
      name: 'Home',
    });
  });

  test('Click on Endpoints link navigates to Endpoints view', () => {
    const endpointsLink = header.root.findByProps({ name: 'Endpoints' });
    endpointsLink.props.onClick();

    expect(navigate.mock.calls.length).toBe(1);
    expect(navigate.mock.calls[0][0]).toEqual({
      name: 'Endpoints',
    });
  });

  test('Click on Logout link triggers firebase auth logout', () => {
    const endpointsLink = header.root.findByProps({ name: 'Logout' });
    endpointsLink.props.onClick();

    expect(signOut.mock.calls.length).toBe(1);
  });
});
