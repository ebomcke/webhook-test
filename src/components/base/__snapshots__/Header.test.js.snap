// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering tests Endpoints link active 1`] = `
<div
  className="ui huge secondary stackable menu"
>
  <div
    className="right menu"
  >
    <a
      className="item"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="active item"
      onClick={[Function]}
    >
      Endpoints
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Logout
    </a>
  </div>
</div>
`;

exports[`Rendering tests Home link active 1`] = `
<div
  className="ui huge secondary stackable menu"
>
  <div
    className="right menu"
  >
    <a
      className="active item"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Endpoints
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Logout
    </a>
  </div>
</div>
`;

exports[`Rendering tests No active link 1`] = `
<div
  className="ui huge secondary stackable menu"
>
  <div
    className="right menu"
  >
    <a
      className="item"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Endpoints
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Logout
    </a>
  </div>
</div>
`;
